cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -530,54 +551,58 @@ func (m *stateMachine) setMountState(part, where string, err error) error {
 	m.degradedState.partition(part).MountLocation = where
 
 	if err := m.verifyMountPoint(where, part); err != nil {
-		m.degradedState.LogErrorf("cannot verify %s mount point at %v: %v",
-			part, where, err)
+		m.degradedState.LogErrorf("cannot verify %s mount point at %v: %v", part, where, err)
 		return err
 	}
 	return nil
 }
 
-func (m *stateMachine) setUnlockStateWithRunKey(partName string, unlockRes secboot.UnlockResult, err error) error {
+func (m *recoverModeStateMachine) setUnlockStateWithRunKey(partName string, unlockRes secboot.UnlockResult, err error) error {
 	part := m.degradedState.partition(partName)
 	// save the device if we found it from secboot
 	if unlockRes.Device != "" {
 		part.FindState = partitionFound
-		part.Device = unlockRes.Device
+		part.rawDevice = unlockRes.Device
+		// if we know the block device, either encrypted or unencrypted, we
+		// start with that as the decryptRawDevice
+		// if the device was actually encrypted and we decrypted, we will
+		// promote decryptRawDevice to the value of the decrypted mapper further
+		// down in this function
+		part.decryptRawDevice = unlockRes.Device
```

Comment:
yes, this looks confusing to me,  it should set from DecryptedDevice (which could be empty)

Reply:
this now uses FsDevice